docs/content/get-started/generate-providers.md

```diff
@@ -0,0 +1,199 @@
+   {{< hint info >}}
+   **Note:** If you don't already have a project to use for testing changes to the Terraform providers, create a project instead of selecting an existing poject. After you finish making your changes, you can delete the project, removing all resources associated with the project.
+   {{< /hint >}}
+   {{< button href="https://console.cloud.google.com/projectselector2/home/dashboard" >}}Go to project selector{{< /button >}}
+1. Make sure that billing is enabled for your Google Cloud project. Learn how to [check if billing is enabled on a project](https://cloud.google.com/billing/docs/how-to/verify-billing-enabled).
+
+## Set up your development environment
+
+{{< hint warning >}}
+If you are familiar with Docker or Podman, you may want to use the experimental [container-based environment]({{< ref "/reference/make-commands.md#container-based-environment" >}}) instead of this section.
+{{< /hint >}}
+
+1. [Install git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)
+1. [Install rbenv](https://github.com/rbenv/rbenv#installation)
+1. Use rbenv to install ruby 3.1.0
+   ```bash
+   rbenv install 3.1.0
+   ```
+1. [Install bundler](https://bundler.io/guides/getting_started.html)
+1. [Install go](https://go.dev/doc/install)
+1. Install goimports
+   ```bash
+   go install golang.org/x/tools/cmd/goimports@latest
+   ```
+1. [Install terraform](https://developer.hashicorp.com/terraform/tutorials/aws-get-started/install-cli)
+1. Clone the `magic-modules` repository
+   ```bash
+   cd ~
+   git clone https://github.com/GoogleCloudPlatform/magic-modules.git
+   ```
+1. Run the following command from the root of your cloned `magic-modules` repository.
+  
+   ```bash
+   cd magic-modules
+   ./scripts/doctor
+   ```
+ 
+   Expected output if everything is installed properly:
+ 
+   ```
+   Check for ruby in path...
+      found!
+   Check for bundle in path...
+      found!
+   Check for go in path...
+      found!
+   Check for goimports in path...
+      found!
+   Check for git in path...
+      found!
+   Check for terraform in path...
+      found!
+   Check for make in path...
+      found!
+   ```
+
+## Generate a provider change
+
+1. In your cloned magic-modules repository, edit `mmv1/products/pubsub/Topic.yaml` to change the description for the schemaSettings field:
+   ```yaml
+   - !ruby/object:Api::Type::NestedObject
+     name: 'schemaSettings'
+     description: |
+       UPDATED_DESCRIPTION
+   ```
+1. Clone the `google` and `google-beta` provider repositories with the following commands:
+
+   ```bash
+   git clone https://github.com/hashicorp/terraform-provider-google.git $GOPATH/src/github.com/hashicorp/terraform-provider-google
+   git clone https://github.com/hashicorp/terraform-provider-google-beta.git $GOPATH/src/github.com/hashicorp/terraform-provider-google-beta
+   ```
+1. Generate changes for the `google` provider
```

Comment:
Should add the link to the section `make commands` here?  I feel the section `make commands` is helpful to generate the providers especially for one product.

Reply:
I'm trying to limit the number of links that might disrupt a user's flow working through the quickstart - see https://cloud.google.com/guides/docs/quickstart for more details. The "`make` commands" reference is linked at the end of the doc for users who want to know more.

Reply:
Got it. Thanks.